src/Translator/TranslatableMessageInterface.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Validator\Translator;
+
+/**
+ * This interface provide methods for extract the translatable parts of a message.
+ */
+interface TranslatableMessageInterface
+{
+    /**
+     * Returns the message to translate.
+     *
+     * The message may include placeholders for message variable interpolation.
+     *
+     * @return string
+     */
+    public function getMessageTemplate();
+
+    /**
+     * Returns the variables to be replaced in the message template placeholders.
+     *
+     * @return array
+     */
+    public function getMessageVariables();
+
+    /**
+     * Returns the translation domain namespace.
+     *
+     * @return string
+     */
+    public function getTranslationDomain();
+
+    /**
+     * Returns the message without translation and with the placeholders replaced by the variables.
```

Comment:
Having messages translated means the validator needs to be translation aware, in order to inject each result with a translator instance, so that you can then get the translated messages. That's a *lot* of pointers running around, for something that is (a) not terribly common, and (b) outside the domain of validation entirely.

It would be simple to create a helper utility that can give back an altered response instance with *translated* messages:

```php
$results = $validator->validate($data);
if (! $results->isValid()) {
    $results = $utility->translate($results); // Returns a new result instance, with translated messages
}
```

We already support message templates in a standard format; the main aspect would be getting that and the message variables to inject.

I'm not convinced we need the translation domain, however; that seems like something that the consumer who wants translated messages would provide at the time they request translated messages.